lib/rouge/lexers/sqf.rb

```diff
@@ -0,0 +1,601 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    class SQF < RegexLexer
+      tag "sqf"
+      filenames "*.sqf"
+
+      title "SQF"
+      desc "Status Quo Function, a Real Virtuality engine scripting language"
+
+      def self.wordoperators
+        @wordoperators = Set.new %w(
+          and or not
+        )
+      end
+
+      def self.initializers
+        @initializers = Set.new %w(
+          private param params
+        )
+      end
+
+      def self.controlflow
+        @controlflow = Set.new %w(
+          if then else exitwith switch do case default while for from to step
+          foreach
+        )
+      end
+
+      def self.constants
+        @constants = Set.new %w(
+          true false player confignull controlnull displaynull grpnull
+          locationnull netobjnull objnull scriptnull tasknull teammembernull
+        )
+      end
+
+      def self.namespaces
+        @namespaces = Set.new %w(
+          currentnamespace missionnamespace parsingnamespace profilenamespace
+          uinamespace
+        )
+      end
+
+      def self.commands
+        @commands = Set.new %w(
+          abs acctime acos action actionids actionkeys actionkeysimages
```

Comment:
How do you make the list?

If it's programmatically generated, add the script as a comment (if it can be a one liner), or add rake task to [tasks/](https://github.com/jneen/rouge/tree/master/tasks).

Reply:
It's generated using the spec/visual/samples/sqf script, with some manual filtering afterwards (because as mentioned in the OP, pretty much everything is a command). Since that is run in game, that's not an option.
Alternatively, there's a wiki page that could be scraped. I will have to check how much it deviates from this list (there might be some outdated commands in there).

Reply:
Could use the same that I used for highlight.js. But it's much simpler since we don't need case sensitivity here.

https://github.com/dedmen/highlight.js_SQFgen

Basically parse
https://github.com/intercept/intercept/blob/master/src/client/headers/client/sqf_assignments.hpp
with regex `\("([^"]*)"`
That list get's regularly refreshed with all commands.